float64 ux
float64 uy
float64 etheta
float64 thetad
float64 uconny
float64 ufy
float64 ucy
float64 uconnx
float64 ufx
float64 ucx
float64 psi